src/Init/Data/Int/Basic.lean

```diff
@@ -333,6 +333,13 @@ instance : Min Int := minOfLe
 
 instance : Max Int := maxOfLe
 
+/--
+Return the absolute value of an integer.
+-/
+def abs : Int → Int
+  | ofNat n   => .ofNat n
+  | negSucc n => .ofNat n.succ
```

Comment:
Can you double-check what exists in mathlib with respect to here, make sure our changes are in-line with mathlib, and document your insights in the PR description?

Reply:
<img width="735" alt="image" src="https://github.com/user-attachments/assets/f574e9fd-2baf-499b-b52e-5931b6ac1b21">

mathlib defines absolute value in the cutest way, as `a U -a` for any group `G` with a lattice structure on it. The lattice structure on `Int` is derived from the order of `<` on it, so this should become equal to our definition, but we cannot reuse the mathlib definition.